llama.cpp

```diff
@@ -15733,6 +15733,22 @@ static int32_t llama_chat_apply_template_internal(
         if (add_ass) {
             ss << "ASSISTANT:";
         }
+    } else if (tmpl == "alpaca" || tmpl.find("### Instruction:\\n") != std::string::npos) {
```

Comment:
Typo? `("### Instruction:\\n")`, looks like an extra \

Reply:
My understanding is that find works on the `chat_template` string, which at least in DeepSeek's case, escapes the newline: https://huggingface.co/deepseek-ai/deepseek-coder-33b-instruct/blob/main/tokenizer_config.json

But let me test it out tomorrow — planning to separate out DeepSeek out in any case.

Reply:
Sorry, I think you're correct. Thanks.